src/states/elder/mod.rs

```diff
@@ -590,26 +609,27 @@ impl Elder {
     // Send NodeApproval to the current candidate which promotes them to Adult and allows them to
     // passively participate in parsec consensus (that is, they can receive gossip and poll
     // consensused blocks out of parsec, but they can't vote yet)
-    fn handle_candidate_approval(&mut self, pub_id: PublicId, outbox: &mut dyn EventBox) {
+    fn handle_candidate_approval(&mut self, p2p_node: P2pNode, _outbox: &mut dyn EventBox) {
         info!(
             "{} Our section with {:?} has approved candidate {}.",
             self,
             self.our_prefix(),
-            pub_id
+            p2p_node
         );
 
+        let pub_id = *p2p_node.public_id();
         let dst = Authority::Node(*pub_id.name());
 
         // Make sure we are connected to the candidate
         if !self.peer_map.has(&pub_id) {
             trace!(
-                "{} - Not yet connected to {} - sending connection request.",
+                "{} - Not yet connected to {} - use p2p_node.",
                 self,
-                pub_id
+                p2p_node
             );
-
-            let src = Authority::Node(*self.name());
-            let _ = self.send_connection_request(pub_id, src, dst, outbox);
+            self.peer_map
+                .insert(pub_id, p2p_node.connection_info().clone());
+            self.send_direct_message(&pub_id, DirectMessage::ConnectionResponse);
```

Comment:
Same here - why do we still need this?

Reply:
Same as previous comment. I agree we want to remove these